source/adaptive_load/metrics_plugin_impl.cc

```diff
@@ -0,0 +1,123 @@
+#include "adaptive_load/metrics_plugin_impl.h"
+
+#include <cmath>
+
+#include "envoy/registry/registry.h"
+
+#include "external/envoy/source/common/common/assert.h"
+#include "external/envoy/source/common/protobuf/protobuf.h"
+
+#include "absl/status/status.h"
+#include "absl/strings/numbers.h"
+
+namespace Nighthawk {
+
+NighthawkStatsEmulatedMetricsPlugin::NighthawkStatsEmulatedMetricsPlugin(
```

Comment:
This is a fairly long function, any way how we can extract some logical sub-portions from it into well named helper functions in an anonymous space? That would help the reader since they can follow the business logic first in this function and dive into the details later.

Reply:
Done.